spec.html

```diff
@@ -34764,78 +34704,50 @@ <h1>%TypedArray%.prototype.reverse ( )</h1>
         <p>This function is not generic. ValidateTypedArray is applied to the *this* value prior to evaluating the algorithm. If its result is an abrupt completion that exception is thrown instead of evaluating the algorithm.</p>
       </emu-clause>
 
-      <emu-clause id="sec-%typedarray%.prototype.set-overloaded-offset">
-        <h1>%TypedArray%.prototype.set ( _overloaded_ [ , _offset_ ] )</h1>
-        <p>%TypedArray%`.prototype.set` is a single function whose behaviour is overloaded based upon the type of its first argument.</p>
+      <emu-clause id="sec-%typedarray%.prototype.set" oldids="sec-%typedarray%.prototype.set-overloaded-offset">
+        <h1>%TypedArray%.prototype.set ( _source_ [ , _offset_ ] )</h1>
+        <p>%TypedArray%`.prototype.set` is a single function whose behaviour differs based upon the type of its first argument.</p>
         <p>This function is not generic. The *this* value must be an object with a [[TypedArrayName]] internal slot.</p>
+        <p>Sets multiple values in this _TypedArray_, reading the values from _source_. The optional _offset_ value indicates the first element index in this _TypedArray_ where values are written. If omitted, it is assumed to be 0.</p>
+        <emu-alg>
+          1. Let _target_ be the *this* value.
+          1. Perform ? RequireInternalSlot(_target_, [[TypedArrayName]]).
+          1. Assert: _target_ has a [[ViewedArrayBuffer]] internal slot.
+          1. Let _targetOffset_ be ? ToInteger(_offset_).
+          1. If _targetOffset_ &lt; 0, throw a *RangeError* exception.
+          1. If _source_ is an Object that has a [[TypedArrayName]] internal slot, then
+            1. Perform ? SetTypedArrayFromTypedArray(_target_, _targetOffset_, _source_).
```

Comment:
Hm. There's a lot of duplicated steps between these two algorithms. Not all of them can be factored out (because of side effects), but a lot of them can: in particular, all of

```
1. Let targetBuffer be target.[[ViewedArrayBuffer]].
1. If IsDetachedBuffer(targetBuffer) is true, throw a TypeError exception.
1. Let targetLength be target.[[ArrayLength]].
1. Let targetName be the String value of target.[[TypedArrayName]].
1. Let targetType be the Element Type value in Table 62 for targetName.
1. Let targetElementSize be the Element Size value specified in Table 62 for targetName.
1. Let targetByteOffset be target.[[ByteOffset]].
```

I think the fact that there's so much duplication inclines me to say these should be inlined, so the duplication can be reduced.

Reply:
Yes, in fact, inlining all the overloads was how I did this one at first. However, the result was about 70 steps long, which I thought was a bit excessive when there were reasonable points at which to delegate to abstract ops.

But I could do the inlining if you'd like to see the result.

Reply:
I'll ask the other editors for opinions at our call later today.

Reply:
Editor group likes the approach you've taken with the new AOs, so let's leave it as-is.